src/main/java/backend/resource/TurboLabel.java

```diff
@@ -211,4 +217,128 @@ public int hashCode() {
+    }
+
+    public static List<String> getLabelsNameList(List<TurboLabel> labels) {
+        return labels.stream()
+                .map(TurboLabel::getActualName)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the list of labels which name contains labelName
+     * @param repoLabels
+     * @param labelName
+     * @return
+     */
+    public static List<String> filterByName(List<String> repoLabels, String labelName) {
+        return repoLabels
+                .stream()
+                .filter(name -> Utility.containsIgnoreCase(getName(name), labelName))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the list of labels which group contains labelGroup
+     * @param repoLabels
+     * @param labelGroup
+     * @return
+     */
+    public static List<String> filterByGroup(List<String> repoLabels, String labelGroup) {
+        if (labelGroup.isEmpty()) return repoLabels;
+
+        return repoLabels
+                .stream()
+                .filter(name -> {
+                    if (hasGroup(name)) {
+                        return Utility.containsIgnoreCase(getGroup(name), labelGroup);
+                    }
+                    return false;
+                })
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the first label that matches the keyword
+     * i.e. the label's group contains keyword's group and label's name contains keyword's name
+     * @param repoLabels
+     * @param keyword
+     * @return
+     */
+    public static String getMatchedLabelName(List<String> repoLabels, String keyword) {
+        List<String> newMatchedLabels = new ArrayList<>();
+        newMatchedLabels.addAll(repoLabels);
+        newMatchedLabels = filterByName(newMatchedLabels, getName(keyword));
+        newMatchedLabels = filterByGroup(newMatchedLabels, getGroup(keyword));
+        return newMatchedLabels.get(0);
+    }
+
+    /**
+     * Returns true if there is exactly 1 matching label for keyword
+     *
+     * A label is matching if:
+     * the label's group contains keyword's group and label's name contains keyword's name
+     * @param repoLabels
+     * @param keyword
+     * @return
+     */
+    public static boolean hasExactlyOneMatchedLabel(List<String> repoLabels, String keyword) {
+        List<String> newMatchedLabels = new ArrayList<>();
+        newMatchedLabels.addAll(repoLabels);
+        newMatchedLabels = filterByName(newMatchedLabels, getName(keyword));
+        newMatchedLabels = filterByGroup(newMatchedLabels, getGroup(keyword));
+        return newMatchedLabels.size() == 1;
+    }
+
+    /**
+     * The labelName is considered to be in an exclusive group if it has a '.' delimiter in it
+     * @param labelName
+     * @return
+     */
+    public static boolean hasExclusiveGroup(String labelName) {
+        return TurboLabel.getDelimiter(labelName).isPresent() && TurboLabel.getDelimiter(labelName).get().equals(".");
```

Comment:
It's not clear what the method `getDelimiter(...)` does. Can you add a header comment to that method?